pkg/ccl/changefeedccl/enriched_source_provider.go

```diff
@@ -134,6 +189,24 @@ func (p *enrichedSourceProvider) KafkaConnectJSONSchema() kcjsonschema.Schema {
 func (p *enrichedSourceProvider) GetJSON(updated cdcevent.Row) (json.JSON, error) {
 	// TODO(#139661): Add other non fixed fields.
 	clear(p.jsonNonFixedData)
+
+	metadata := updated.Metadata
+	tableID := metadata.TableID
+	tableInfo, ok := p.sourceData.tableSchemaInfo[tableID]
+	if !ok {
+		return nil, fmt.Errorf("table %d not found in tableSchemaInfo", tableID)
+	}
+
+	p.jsonNonFixedData[fieldNameDatabaseName] = json.FromString(tableInfo.dbName)
+	p.jsonNonFixedData[fieldNameSchemaName] = json.FromString(tableInfo.schemaName)
+	p.jsonNonFixedData[fieldNameTableName] = json.FromString(tableInfo.tableName)
+
+	primaryKeysBuilder := json.NewArrayBuilder(len(tableInfo.primaryKeys))
```

Comment:
nit: might be worth storing this info as json ahead of time. ditto the above, but `json.FromString` is basically a no-op so i dont think that's necessary